ncviewjs_backend/rechunking/rechunk.py

```diff
@@ -0,0 +1,135 @@
+    transfer_str = f"skyplane cp -r -y {staging_store} {prod_store}"
+    print(transfer_str)
+    subprocess.check_output(transfer_str, shell=True)
+
+
+def dataset_is_valid(*, zarr_store_url: pydantic.HttpUrl):
+    """Checks that rechunked dataset is valid."""
+    try:
+        with xr.open_dataset(zarr_store_url, engine='zarr', chunks={}, decode_cf=False):
+            pass
+    except Exception as exc:
+        raise RuntimeError(f'Error opening rechunked dataset: {zarr_store_url}') from exc
+
+
+def rechunk_dataset(
+    *,
+    zarr_store_url: str,
+    cf_axes_dict: dict,
+    store_paths: dict,
+    spatial_chunk_square_size: int = 256,
+    target_size_bytes: int = 5e5,
+    max_mem: str = "1000MB",
+):
+    """Rechunks zarr dataset to match chunk schema required by carbonplan web-viewer.
+
+    Parameters
+    ----------
+    zarr_store_url : str
+        Input zarr store url to be rechunked
+    cf_dims_dict : dict
+        dictionary of CF dims
+    store_paths : dict
+        rechunker temp and target paths
+    spatial_chunk_square_size : int, optional
+        X,Y spatial chunk size by default 256
+    target_size_bytes : int, optional
+        target size in bytes, by default 5e5
+    max_mem : str, optional
+        max memory available for rechunking operation, by default "1000MB"
+
+    Returns
+    -------
+    str
+        path to rechunked target store
+    """
+
+    time_chunk = determine_chunk_size(
+        spatial_chunk_square_size=spatial_chunk_square_size, target_size_bytes=target_size_bytes
+    )
+
+    ds = xr.open_dataset(zarr_store_url, engine='zarr', chunks={}, decode_cf=False)
+    print(ds)
+    group = zarr.open_consolidated(zarr_store_url)
+
+    chunks_dict = {}
+    for variable, axes in cf_axes_dict.items():
+        variable_dims = ds[variable].dims
+        sizes = ds[variable].sizes
+
+        result = {}
+        for key, value in axes.items():
+            if key == 'T':
+                result[value] = min(time_chunk, sizes[value])
+            elif key in ['X', 'Y']:
+                result[value] = min(spatial_chunk_square_size, sizes[value])
+        for dim in variable_dims:
+            if dim not in result.keys():
+                result[dim] = -1
+
+        chunks_dict[variable] = result
+
+    print(f'Chunks: {chunks_dict}')
+
+    tmp_mapper = fsspec.get_mapper(store_paths['temp_store'])
+    tgt_mapper = fsspec.get_mapper(store_paths['staging_store'])
+
+    array_plan = rechunker.rechunk(group, chunks_dict, max_mem, tgt_mapper, temp_store=tmp_mapper)
+    array_plan.execute()
+
+    # Consolidate metadata
+    zarr.consolidate_metadata(store_paths['staging_store'])
+
+    # Check that rechunked dataset is valid
+    dataset_is_valid(zarr_store_url=store_paths['staging_store'])
+
+
+def generate_stores(*, key: str, bucket: str, md5_id: str):
+    settings = get_settings()
+    # TODO: use a better naming scheme
+    store_suffix = f"{uuid.uuid1()}-{md5_id}.zarr"
```

Comment:
@norlandrhagen, feel free to set this to the original value you had at the start of this PR.